### PR TITLE
Feat: add BigInt serialization support

### PR DESCRIPTION
**Pull request by antho1404**
Adds support for serializing and deserializing JavaScript bigint values in workflows.

## Changes
- Add bigint reducer to serialize bigint to string (via toString())
- Add bigint reviver to deserialize string back to BigInt()
- Add bigint to the Serializable type in schemas
- Document bigint support
- Add tests for positive and negative bigint values

Note: I use bigint for blockchain-related apps where every price is precise to up to 18 decimals which becomes higher than `Number.MAX_SAFE_INTEGER` quickly

### CODE REVIEW

File: packages/core/src/serialization.ts [hunk 1]
```diff
@@ -474,6 +476,7 @@ function getCommonRevivers(global: Record<string, any> = globalThis) {
   }
   return {
     ArrayBuffer: reviveArrayBuffer,
+    BigInt: (value: string) => BigInt(value),
```
Comment (antho1404) [inline packages/core/src/serialization.ts]: Does this require the global type? This part is a bit unclear, if yes, should BigInt be added to this context somewhere else?

```suggestion
    BigInt: (value: string) => global.BigInt(value),
```

File: .changeset/bigint-serialization.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@workflow/core": patch
+---
+
+Add missing support for bigint serialization
```
Comment (TooTallNate) [inline .changeset/bigint-serialization.md]:
```suggestion
Add support for bigint serialization
```

File: packages/core/src/serialization.ts (hunk 1 above)
Comment (TooTallNate) [reply packages/core/src/serialization.ts]: To answer your question, yes, generally speaking the constructors from the provided `globalThis` should be used. This is because the (de)serialization happens outside of the workflow VM context, but we might be rehydrating an object that will be used inside the VM, in which case the correct constructor needs to be used for `instanceof` to work.

For this specific case it might not be necessary because `bigint` is a primitive type. But I applied your change here for consistency.

**Comment by TooTallNate** [comment]
Thank you!
